lib/std/Url.zig

```diff
@@ -1,98 +1,519 @@
+        "1.2.3.4",
+        "192.168.0.1",
+        "10.42.0.0",
+    });
+}
+
+test "authority.IPv6" {
+    try testAuthorityHost(.{
+        "[2001:db8:0:0:0:0:2:1]",
+        "[2001:db8::2:1]",
+        "[2001:db8:0000:1:1:1:1:1]",
+        "[2001:db8:0:1:1:1:1:1]",
+        "[0:0:0:0:0:0:0:0]",
+        "[0:0:0:0:0:0:0:1]",
+        "[::1]",
+        "[::]",
+        "[2001:db8:85a3:8d3:1319:8a2e:370:7348]",
+        "[fe80::1ff:fe23:4567:890a%25eth2]",
+        "[fe80::1ff:fe23:4567:890a]",
+        "[fe80::1ff:fe23:4567:890a%253]",
+        "[fe80:3::1ff:fe23:4567:890a]",
+    });
+}
+
+test "RFC example 1" {
+    const uri = "foo://example.com:8042/over/there?name=ferret#nose";
+    try std.testing.expectEqual(Url{
+        .scheme = uri[0..3],
+        .user = null,
+        .password = null,
+        .host = uri[6..17],
+        .port = 8042,
+        .path = uri[22..33],
+        .query = uri[34..45],
+        .fragment = uri[46..50],
+    }, try parse(uri));
+}
+
+test "RFX example 2" {
+    const uri = "urn:example:animal:ferret:nose";
+    try std.testing.expectEqual(Url{
+        .scheme = uri[0..3],
+        .user = null,
+        .password = null,
+        .host = null,
+        .port = null,
+        .path = uri[4..],
+        .query = null,
+        .fragment = null,
+    }, try parse(uri));
+}
+
+// source:
+// https://en.wikipedia.org/wiki/Uniform_Resource_Identifier#Examples
+test "Examples from wikipedia" {
+    // these should all parse
+    const list = [_][]const u8{
+        "https://john.doe@www.example.com:123/forum/questions/?tag=networking&order=newest#top",
+        "ldap://[2001:db8::7]/c=GB?objectClass?one",
+        "mailto:John.Doe@example.com",
+        "news:comp.infosystems.www.servers.unix",
+        "tel:+1-816-555-1212",
+        "telnet://192.0.2.16:80/",
+        "urn:oasis:names:specification:docbook:dtd:xml:4.1.2",
+        "http://a/b/c/d;p?q",
+    };
+    for (list) |uri| {
+        _ = try parse(uri);
+    }
+}
+
+// source:
+// https://tools.ietf.org/html/rfc3986#section-5.4.1
+test "Examples from RFC3986" {
+    // these should all parse
+    const list = [_][]const u8{
+        "http://a/b/c/g",
+        "http://a/b/c/g",
+        "http://a/b/c/g/",
+        "http://a/g",
+        "http://g",
+        "http://a/b/c/d;p?y",
+        "http://a/b/c/g?y",
+        "http://a/b/c/d;p?q#s",
+        "http://a/b/c/g#s",
+        "http://a/b/c/g?y#s",
+        "http://a/b/c/;x",
+        "http://a/b/c/g;x",
+        "http://a/b/c/g;x?y#s",
+        "http://a/b/c/d;p?q",
+        "http://a/b/c/",
+        "http://a/b/c/",
+        "http://a/b/",
+        "http://a/b/",
+        "http://a/b/g",
+        "http://a/",
+        "http://a/",
+        "http://a/g",
+    };
+    for (list) |uri| {
+        _ = try parse(uri);
+    }
+}
+
+test "Special test" {
+    // This is for all of you code readers ♥
+    _ = try parse("https://www.youtube.com/watch?v=dQw4w9WgXcQ&feature=youtu.be&t=0");
+}
+
```

Comment:
```suggestion
```

Reply:
Not sure if i want to remove that. It's a very important URL

Reply:
Rick-rolled by Nixpkgs! OH NO!